### ISSUE TITLE
Document all flags associated with manager

### ISSUE THREAD

**Issue by atoulme** [open-telemetry/opentelemetry-operator#3565]
### Component(s)

auto-instrumentation

### Is your feature request related to a problem? Please describe.

The manager executable seems to accept a number of ways of configuration:
* A JSON file
* CLI flags
* Env vars

There is no uniformity between those arguments - some env vars for example like ENABLE_WEBHOOKS have no equivalent CLI flag.

### Describe the solution you'd like

* Create a documentation for the existing
* Create tests to uniformly enforce how the config is set so each key is sure to have an equivalent in JSON/CLI Flag/Env var
* Fix inconsistencies

**Comment by jaronoff97** [comment]
Agreed, this is in a few places currently with most of documentation being the code itself. This would be a good one to take.

**Comment by shuheiktgw** [comment]
Let me work on this issue as my first contribution to the project, if I may 🙏

**Comment by swiatekm** [comment]
After @atoulme's recent refactor of configuration loading, this should now be possible looking solely at the [config package.](https://github.com/open-telemetry/opentelemetry-operator/tree/main/internal/config). I'd create a `docs/configuration.md`, then remove all configuration references from the README and replace them with links to the new document.

**Comment by siddhant-khisty** [comment]
Hey, I'd like to work on this

**Comment by gaurav337** [comment]
Hi, I’d like to work on this. I’ll create a new PR updating docs/configuration.md to include CLI flags, JSON options, env vars, and defaults, addressing the feedback from #4309.

**Comment by swiatekm** [comment]
@gaurav337 could you base your change on https://github.com/open-telemetry/opentelemetry-operator/pull/5259? Part of a recently merged docs RFC is also to generate this documentation automatically.

**Comment by gaurav337** [comment]
Hi @swiatekm,
I’m ready to tackle the auto-generated config docs for #3565.
Since #5259 introduces the new reference/ structure, I want to ensure my PR aligns with it. Should I base my branch on swiatekm:docs/reorganize to get the correct folder structure now, or would you prefer I wait until #5259 merges into main?
Thanks!

**Comment by swiatekm** [comment]
@gaurav337 it's simpler if you base you branch on mine, I think. It's unlikely there's going to be sweeping changes in it, so once it's merged, your PR should apply cleanly to main.

**Comment by amarkdotdev** [comment]
Opened #5305 with a manager configuration reference (`docs/reference/configuration.md`) covering YAML file, env, and CLI settings, plus a fix for `--tls-min-version` CLI parsing and a unit test.

**Comment by swiatekm** [comment]
Reopening because #5305 doesn't auto-generate the docs. That is still pending.
